types/mydecimal.go

```diff
@@ -1383,6 +1383,20 @@ func (d *MyDecimal) Compare(to *MyDecimal) int {
 	return 1
 }
 
+// DecimalNeg reverses decimal's sign.
+func DecimalNeg(from *MyDecimal) *MyDecimal {
+	to := *from
+	if from.IsZero() {
+		return &to
+	}
+	if from.IsNegative() {
```

Comment:
How about `to.negative = !from.negative`?

Reply:
done~